src/Tree/Mapping/Validator.php

```diff
@@ -157,7 +157,7 @@ public function isValidFieldForLockTime($meta, $field)
     {
         $mapping = $meta->getFieldMapping($field);
 
-        return $mapping && ('date' === $mapping['type'] || 'datetime' === $mapping['type'] || 'timestamp' === $mapping['type']);
+        return $mapping && ('date' === ($mapping->type ?? $mapping['type']) || 'datetime' === ($mapping->type ?? $mapping['type']) || 'timestamp' === ($mapping->type ?? $mapping['type']));
```

Comment:
Can we just do a single `$type = $mapping->type ?? $mapping['type']` here instead of repeating it for every supported datetime field?  😅